core/src/avm2/globals/flash/net.rs

```diff
@@ -15,6 +17,47 @@ pub mod socket;
 pub mod url_loader;
 pub mod xml_socket;
 
+fn object_to_index_map<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    obj: &Object<'gc>,
+) -> Result<IndexMap<String, String>, Error<'gc>> {
+    let mut map = IndexMap::new();
+    let mut last_index = obj.get_next_enumerant(0, activation)?;
+    while let Some(index) = last_index {
+        let name = obj
+            .get_enumerant_name(index, activation)?
+            .coerce_to_string(activation)?;
+        let value = obj
+            .get_public_property(name, activation)?
+            .coerce_to_string(activation)?
+            .to_utf8_lossy()
+            .to_string();
+
+        let name = name.to_utf8_lossy().to_string();
+        map.insert(name, value);
+        last_index = obj.get_next_enumerant(index, activation)?;
+    }
+    Ok(map)
+}
+
+fn parse_data<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    url: &String,
+    data: &Value<'gc>,
+) -> Result<(String, IndexMap<String, String>), Error<'gc>> {
+    let mut url = url.to_string();
+    let obj = data.coerce_to_object(activation)?;
+    let vars = object_to_index_map(activation, &obj).unwrap_or_default();
+    let str_data = data.coerce_to_string(activation)?.to_string();
+    if vars.is_empty() && !str_data.is_empty() {
+        if !url.contains('?') && !str_data.contains('?') {
```

Comment:
You are basically re-implementing https://github.com/ruffle-rs/ruffle/blob/master/core/src/avm2/globals/flash/display/loader.rs#L172 now.

In my local testing your `!str_data.contains('?')` check while somewhat logically, doesn't actually happen in Flash.

Reply:
And this can't fully use the loader.rs implementation because of the backend's call signature.